test/search_api/test_search_api_query_filter_factory.py

```diff
@@ -0,0 +1,2098 @@
+                        "limit": 50,
+                        "skip": 20,
+                    },
+                },
+                "datasets",
+                4,
+                [[], ["instrument"], [], []],
+                [["title", "eq", "My Title", "and"], [], [], []],
+                [None, None, 50, None],
+                [None, None, None, 20],
+                id="Simple case",
+            ),
+            pytest.param(
+                {
+                    "filter": {
+                        "where": {
+                            "and": [
+                                {
+                                    "and": [
+                                        {"summary": "My Test Summary"},
+                                        {"title": {"like": "Test title"}},
+                                    ],
+                                },
+                                {
+                                    "and": [
+                                        {"pid": "Test pid"},
+                                        {"type": {"eq": "Test type"}},
+                                    ],
+                                },
+                                {
+                                    "or": [
+                                        {"doi": "Test doi"},
+                                        {"license": {"like": "Test license"}},
+                                    ],
+                                },
+                            ],
+                        },
+                        "include": [
+                            {
+                                "relation": "instrument",
+                                "scope": {
+                                    "where": {"name": "Instrument 1"},
+                                    "limit": 2,
+                                },
+                            },
+                        ],
+                        "limit": 50,
+                        "skip": 20,
+                    },
+                },
+                "datasets",
+                11,
+                [[], [], [], [], [], [], ["instrument"], [], [], [], []],
+                [
+                    ["summary", "eq", "My Test Summary", "and"],
+                    ["title", "like", "Test title", "and"],
+                    ["pid", "eq", "Test pid", "and"],
+                    ["type", "eq", "Test type", "and"],
+                    ["doi", "eq", "Test doi", "or"],
+                    ["license", "like", "Test license", "or"],
+                    [],
+                    ["instrument.name", "eq", "Instrument 1", "and"],
+                    [],
+                    [],
+                    [],
+                ],
+                [None, None, None, None, None, None, None, None, 2, 50, None],
+                [None, None, None, None, None, None, None, None, None, None, 20],
+                id="Complex case",
+            ),
```

Comment:
In this test case, we end up with two `SearchAPILimitFilter` objects but how do we know which one is for the `datasets` entity and which one is for the `instrument` relation? I do not think we can work this out. Maybe the `SearchAPILimitFilter` class should have another field called `entity_name` or something else that will indicate to which entity the LIMIT filter is related to?

The same applies to SKIP filters.

Reply:
We decided not to support LIMIT and SKIP filters on relations so marking this as resolved.